src/shared/cgroup-setup.c

```diff
@@ -45,6 +106,10 @@ bool cg_is_unified_wanted(void) {
         if (r > 0 && streq_ptr(c, "all"))
                 return (wanted = true);
 
+        /* If we are in a container and it looks like unified isn't possible, don't use it */
+        if (detect_container() > 0 && !cg_is_unified_possible())
```

Comment:
I think the detect_container8) check should go. even if we run on the host: if a controller has been mounted in v1 mode we should not go for unified mode

Reply:
sounds good, thanks